### PR TITLE
Effects and Potions Guides

### PR DESCRIPTION
**Pull request by dicedpixels**
Ports the [Adding Status Effects](https://fabricmc.net/wiki/tutorial:status_effects) and [Adding Potions](https://fabricmc.net/wiki/tutorial:potions) guides.

### CODE REVIEW

**Comment by dicedpixels** [comment]
Welp, I'm not sure why the build is failing. The assets are right there 😤.

File: develop/effects.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+title: Effects
+description: Adding a custom effect.
```
Comment (IMB11) [inline develop/effects.md]:
```suggestion
description: Learn how to add custom status effects.
```

File: develop/potions.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
+title: Potions
+description: Adding a custom potion.
```
Comment (IMB11) [inline develop/potions.md]:
```suggestion
description: Learn how to add a custom potion for various status effects.
```

**Comment by IMB11** [comment]
> Welp, I'm not sure why the build is failing. The assets are right there 😤.

`Could not resolve "./../../assets/develop/tater-potion.png" from "develop/potions.md"`

**Comment by dicedpixels** [comment]
> > Welp, I'm not sure why the build is failing. The assets are right there 😤.
> 
> `Could not resolve "./../../assets/develop/tater-potion.png" from "develop/potions.md"`

Well, I'm blind. Fixed.

File: develop/potions.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Adding a potion follows a similar path as adding an item. You will create an instance of your potion and register it by
+calling `BrewingRecipeRegistry.registerPotionRecipe`.
+
+::: info
+When Fabric API is present, `BrewingRecipeRegistry.registerPotionRecipe` is made accessible through an Access Widener.
+
+You can directly call `BrewingRecipeRegistry.registerPotionRecipe` without using a mixin.
+:::
+
+Let's assume you don't have Fabric API and instead rely on a mixin invoker to access
```
Comment (IMB11) [inline develop/potions.md]: I think this should be avoided - it'll cause some confusion and people definitely skim read the callout and miss that this bit is completely optional.

Instead you should assume developers have Fabric API and put a callout briefly explaining how to do it without, eg: "If you do not have Fabric API, you can create a mixin invoker, or use access wideners."

File: develop/potions.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+We pass an instance of `StatusEffectInstance`, which takes 3 parameters:
+
+* `StatusEffect type` - An effect. We use our custom effect here. Alternatively you can access vanilla effects
+  through `net.minecraft.entity.effect.StatusEffects`.
+* `int duration` - Duration of the effect in game ticks.
+* `int amplifier` - An amplifier for the effect. For example, Haste II would have an amplifier of 1.
+
+### Registering the Potion
+
+In our initializer, we call `BrewingRecipeRegistry.registerPotionRecipe` using our mixin invoker.
```
Comment (IMB11) [inline develop/potions.md]: See above comment about Fabric API assumption.

File: develop/effects.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+Let's create a custom effect class by extending `StatusEffect`, which is the base class for all effects.
+
+@[code lang=java transcludeWith=:::1](@/reference/latest/src/main/java/com/example/docs/effect/TaterEffect.java)
+
+### Registering Your Custom Effect
+
+Similar to block and item registration, we use `Registry.register` to register our custom effect into the
+`STATUS_EFFECT` registry. This can be done in our initializer.
+
+@[code lang=java transcludeWith=:::1](@/reference/latest/src/main/java/com/example/docs/effect/FabricDocsReferenceEffects.java)
```
Comment (IMB11) [inline develop/effects.md:36]: Might want to add a callout that links to the "Potions" section, eg: "If you want to add a custom potion item for your effect, checkout the [Potions] page."

**Review by IMB11 — CHANGES_REQUESTED** [review]
Just some small things to take a look at, overall it's good :+1:

File: develop/potions.md (hunk 3 above)
Comment (dicedpixels) [reply develop/potions.md]: I moved the mixin accessor route to a separate section. Main example now uses the FAPI approach.

File: develop/potions.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+::: info
+When Fabric API is present, `BrewingRecipeRegistry.registerPotionRecipe` is made accessible through an Access Widener.
+:::
+
+### Creating the Potion
+
+Let's start by declaring a field to store your `Potion` instance. We will be directly using the initializer class to
+hold this.
+
+@[code lang=java transclude={20-28}](@/reference/latest/src/main/java/com/example/docs/potion/FabricDocsReferencePotions.java)
```
Comment (IMB11) [inline develop/potions.md]: Seems to be the wrong transclude line numbers?

image

File: develop/potions.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+::: info
+To create your own effect, please see the [Effects](./effects.md) guide.
+:::
+
+### Registering the Potion
+
+In our initializer, we call `BrewingRecipeRegistry.registerPotionRecipe`.
+
+@[code lang=java transclude={31-31}](@/reference/latest/src/main/java/com/example/docs/potion/FabricDocsReferencePotions.java)
```
Comment (IMB11) [inline develop/potions.md]: Same here - wrong transclude line numbers?

image

**Review by IMB11 — CHANGES_REQUESTED** [review]
Just some minor changes, then it should be ready to merge.

**Review by IMB11 — APPROVED** [review]

**Pull request merged**